rm-datasource/src/main/java/io/seata/rm/datasource/exec/SelectForUpdateExecutor.java

```diff
@@ -90,16 +105,25 @@ public T doExecute(Object... args) throws Throwable {
                     }
                     break;
                 } catch (LockConflictException lce) {
-                    conn.rollback(sp);
+                    //keep same logic as above
+                    if (originalAutoCommit) {
+                        conn.setAutoCommit(true);
+                        conn.rollback();
+                    } else if (!originalAutoCommit && dbmd.supportsSavepoints()) {
+                        conn.rollback(sp);
+                    }
                     lockRetryController.sleep(lce);
                 }
             }
         } finally {
             if (sp != null) {
```

Comment:
Missing logic to restore `originalAutoCommit`.

Reply:
done, thank you.